platform/core/src/utils/combineFrameInstance.ts

```diff
@@ -9,34 +11,68 @@
  * single frame data. (eg frame is undefined is the same as frame===1).
  */
 const combineFrameInstance = (frame, instance) => {
-  const { PerFrameFunctionalGroupsSequence, SharedFunctionalGroupsSequence, NumberOfFrames } =
-    instance;
+  const {
+    PerFrameFunctionalGroupsSequence,
+    SharedFunctionalGroupsSequence,
+    NumberOfFrames,
+    SpacingBetweenSlices,
+  } = instance;
 
   if (PerFrameFunctionalGroupsSequence || NumberOfFrames > 1) {
     const frameNumber = Number.parseInt(frame || 1);
-    const shared = (
-      SharedFunctionalGroupsSequence ? Object.values(SharedFunctionalGroupsSequence[0]) : []
-    )
-      .filter(it => !!it)
-      .map(it => it[0])
-      .filter(it => it !== undefined && typeof it === 'object');
-    const perFrame = (
-      PerFrameFunctionalGroupsSequence
-        ? Object.values(PerFrameFunctionalGroupsSequence[frameNumber - 1])
-        : []
-    )
-      .filter(it => !!it)
-      .map(it => it[0])
-      .filter(it => it !== undefined && typeof it === 'object');
+    const shared = SharedFunctionalGroupsSequence
```

Comment:
This is ok here, but I'd like to discuss with you the option of making it use the javascript objects with inheritance - we can start with an object containing all the instance data, which we have, then use
const shared = Object.create(instance, ... shared functional groups)
SharedFunctionalGroupsSequence. <hidden attribute> = shared
Then for the functional groups, use a hidden value per frame which is
Object.create(shared, ...per frame values)
That is going to be far more memory efficient, and will avoid recreating the objects all the time.

Reply:
Yeah i have opened an issue to implement it https://github.com/OHIF/Viewers/issues/4551